core/chains/evm/log/registrations.go

```diff
@@ -108,14 +140,36 @@ func (r *registrations) removeSubscriber(sub *subscriber) (needsResubscribe bool
 
 	needsResubscribe = handlers.removeSubscriber(sub)
 
-	if len(r.handlersByConfs[sub.opts.MinIncomingConfirmations].handlersByAddr) == 0 {
+	if len(r.handlersByConfs[sub.opts.MinIncomingConfirmations].lookupSubs) == 0 {
 		delete(r.handlersByConfs, sub.opts.MinIncomingConfirmations)
 		r.resetHighestNumConfirmationsValue()
 	}
 
 	return
 }
 
+// checkRemoveSubscriber validates we aren't violating any assumptions by removing this subscriber
+func (r *registrations) checkRemoveSubscriber(sub *subscriber) error {
```

Comment:
The function name suggests it is only checking assumptions. But this function is also actually deleting stuff.
I would suggest, if this function returns success, then do all delete operations together.

Or if the deletes are really needed here with validations, atleast rename the function to be more clear.

Reply:
It only modifies maps used by the check function to determine invariant violations. The data structures are only used for the checks and nothing else.